### DISCUSSION TITLE
Information on supporting compression in zlib.a

### DISCUSSION THREAD

**Discussion by rsbeckerca** [openssl/openssl#22018] [Q&A] [answered]
Now that compression is supported at 3.2, I would like to be able to add compression support to the NonStop builds. The location is not standard (like `/usr/lib`), but `/usr/coreutils/lib`. In addition, we have both `libz.so` and `libz.a`. For compatibility with OS versions more than 2 years old, I need to use `libz.a`. How can I configure the builds to use that?

**Comment by paulidale** [comment]
I'm not sure how to force `libz.a` without doing a static build.
For the paths there are a [couple of configuration options](https://github.com/openssl/openssl/blob/master/INSTALL.md#with-zlib-include).

**Reply by rsbeckerca** [reply]
As long as I can specify, somewhere, the /usr/coreutils/lib/libz.a file in the link instructions prior to -L, it should make it. Just not sure how in the configuration infrastructure I'm ok with doing that outside the standard recipe.

**Comment by rsbeckerca** [comment, marked as answer]
@paulidale I managed to get the code to configure properly but I am having a memory model argument with libz.a. For later:
```
LDFLAGS=/usr/coreutils/lib/libz.a ./Configure nonstop-nsx_64 --with-rand-seed=rdcpu enable-fips enable-zlib --with-zlib-lib=/usr/coreutils/lib --with-zlib-include=/usr/coreutils/include
```
Close the discussion as resolved.
